pyee/_trio.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+
+from contextlib import asynccontextmanager
+import trio
+from pyee._base import BaseEventEmitter, PyeeException
+
+__all__ = ['TrioEventEmitter']
+
+
+class TrioEventEmitter(BaseEventEmitter):
+    """An even emitter class which can run trio coroutines in a trio nursery.
+
+    By default, this class will lazily create both a nursery manager (the
+    object returned from ``trio.open_nursery()`` and a nursery (the object
+    yielded by using the nursery manager as an async context manager). It is
+    also possible to supply an existing nursery manager via the ``manager``
+    argument, or an existing nursery via the ``nursery`` argument.
+
+    Instances of TrioEventEmitter are themselves async context managers, so
+    that they may manage the lifecycle of the underlying trio nursery. For
+    example, typical usage of this library may look something like this::
+
+        async with TrioEventEmitter() as ee:
+            # Underlying nursery is instantiated and ready to go
+            @ee.on('data')
+            async def handler(data):
+                print(data)
+
+            ee.emit('event')
+
+        # Underlying nursery and manager have been cleaned up
+
+    Unlike the case with the BaseEventEmitter, all exceptions raised by event
+    handlers are automatically emitted on the ``error`` event. This is
+    important for trio coroutines specifically but is also handled for
+    synchronous functions for consistency.
+
+    For trio coroutine event handlers, calling emit is non-blocking. In other
+    words, you should not attempt to await emit; the coroutine is scheduled
+    in a fire-and-forget fashion.
+
+    Note that this behavior is a little against the grain for typical trio use.
```

Comment:
Because? Might be worth adding another sentence about callbacks and design etc.

Reply:
Yeah, low key thinking I should remove this comment and add a more broad disclaimer to the introduction in a separate PR.